src/utils/remark/with-prose.js

```diff
@@ -0,0 +1,53 @@
+/*
+  Usage:
+
+  const withMDX = require("@next/mdx")({
+    options: {
+      remarkPlugins: [
+        withProse,
+      ],
+    },
+  });
+*/
+
+const proseComponents = ['Heading']
+
+const isJsNode = (node) => {
+  return (
+    ['jsx', 'import', 'export'].includes(node.type) &&
+    !/^<[a-z]+(>|\s)/.test(node.value) &&
+    !new RegExp(`^<(${proseComponents.join('|')})(>|\\s)`).test(node.value)
+  )
+}
+
+module.exports.withProse = () => {
+  return (tree) => {
+    console.log('WITH PROSE')
+    let insideProse = false
+    tree.children = tree.children.flatMap((node, i) => {
+      console.log('PROSE', insideProse, isJsNode(node), node)
+      if (insideProse && isJsNode(node)) {
+        insideProse = false
+        return [{type: 'jsx', value: '</div>'}, node]
```

Comment:
I had to replace some `section` elements we were using in articles with `div` for the build to pass since it's only expecting divs. That seems like something that could be improved, otherwise we'll just have to remember it and use divs.